pkg/workloads/k8s.go

```diff
@@ -324,40 +388,42 @@ func GenerateNodeAffinities(apiNodeGroups []string) *kcore.Affinity {
 	}
 }
 
-func RealtimeProxyContainer(api spec.API) (kcore.Container, kcore.Volume) {
-	return kcore.Container{
-		Name:            _proxyContainerName,
-		Image:           config.ClusterConfig.ImageProxy,
-		ImagePullPolicy: kcore.PullAlways,
-		Args: []string{
-			"-port",
-			consts.ProxyListeningPortStr,
-			"-metrics-port",
-			consts.MetricsPortStr,
-			"-user-port",
-			s.Int32(*api.Pod.Port),
-			"-max-concurrency",
-			s.Int32(int32(api.Autoscaling.MaxConcurrency)),
-			"-max-queue-length",
-			s.Int32(int32(api.Autoscaling.MaxQueueLength)),
-			"-cluster-config",
-			consts.DefaultInClusterConfigPath,
-		},
-		Ports: []kcore.ContainerPort{
-			{Name: "metrics", ContainerPort: consts.MetricsPortInt32},
-			{ContainerPort: consts.ProxyListeningPortInt32},
-		},
-		Env: []kcore.EnvVar{
-			{
-				Name:  "CORTEX_LOG_LEVEL",
-				Value: strings.ToUpper(userconfig.InfoLogLevel.String()),
-			},
-		},
-		EnvFrom: baseClusterEnvVars(),
-		VolumeMounts: []kcore.VolumeMount{
-			ClusterConfigMount(),
-		},
-	}, ClusterConfigVolume()
+// Generate map from container name -> probe
+func GenerateProbesConfigMapData(probes map[string]kcore.Probe) (map[string]string, error) {
+	probesEncoded, err := json.MarshalIndent(probes, "", "\t")
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]string{
+		"probes.json": string(probesEncoded),
+	}, nil
+}
+
+func GenerateJobSpecConfigMapData(taskJobSpec *spec.TaskJob, batchJobSpec *spec.BatchJob) (map[string]string, error) {
```

Comment:
as David mentioned before, these should be mounted in the same config map at once, so these helper functions can probably be removed